tests/test_integration.py

```diff
@@ -575,6 +575,33 @@ def test_interpreter_resolution_pex_python_path_precedence_over_pex_python():
         assert correct_interpreter_path in stdout
 
 
+def test_use_first_matching_interpreter():
+    with temporary_dir() as output_dir:
+        py35_path = ensure_python_interpreter(PY35)
+        py36_path = ensure_python_interpreter(PY36)
+        env = make_env(
+            PEX_IGNORE_RCFILES="1",
+            PATH=os.pathsep.join([os.path.dirname(py35_path), os.path.dirname(py36_path)]),
+        )
+        pex_out_path = os.path.join(output_dir, "pex_py2.pex")
+        res = run_pex_command(
+            [
+                "--disable-cache",
+                "--interpreter-constraint=>=3.5",
+                "--use-first-matching-interpreter",
+                "-v",
+                "-o",
+                pex_out_path,
+            ], env=env
+        )
+        res.assert_success()
+        assert "using {}".format(py35_path) in res.error.decode()
+        assert "will not be used: {}".format(py36_path) in res.error.decode()
```

Comment:
This isn't as comprehensive as I'd like. All this verifies is that we're logging, but doesn't actually verify the behavior is correct.

An alternative idea is to build with a Pex that has distinct wheels for each Python version, e.g. `cp35m` and `cp36m` wheels. Then, try running that Pex with both Py35 and Py36. Py36 should fail. If we wanted to be really robust, we could do the same thing without `--use-first-matching-interpreters` and ensure that both Py35 and Py36 work that time.